<!DOCTYPE html> <html> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> What are GO, MSigDB, KEGG, ORA, and GSEA? | The Whispers of Biology </title> <meta name="author" content="Krittiyabhorn (Namthip) Kongtanawanich"> <meta name="description" content="What are those terms in transcriptomics analysis?"> <meta name="keywords" content="Namthip, Kuchiki Namthip, The Whispers of Biology, omic"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" href="/assets/css/scholar-icons.css?62b2ac103a88034e6882a5be5f3e2772"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%8C%B8&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://kuchikinamthip.github.io/blog/2024/msigDB_ORA_GSEA/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>
    initTheme();
  </script> <script src="/assets/js/distillpub/template.v2.js"></script> <script src="/assets/js/distillpub/transforms.v2.js"></script> <script src="/assets/js/distillpub/overrides.js"></script> </head> <body> <d-front-matter> <script async type="text/json">
      {
            "title": "What are GO, MSigDB, KEGG, ORA, and GSEA?",
            "description": "What are those terms in transcriptomics analysis?",
            "published": "September 18, 2024",
            "authors": [
              
              {
                "author": "Krittiyabhorn (Namthip) Kongtanawanich",
                "authorURL": "https://kuchikinamthip.github.io/",
                "affiliations": [
                  {
                    "name": "Faculty of Medicine Siriraj Hospital, Mahidol University, Thailand",
                    "url": ""
                  }
                ]
              }
              
            ],
            "katex": {
              "delimiters": [
                {
                  "left": "$",
                  "right": "$",
                  "display": false
                },
                {
                  "left": "$$",
                  "right": "$$",
                  "display": true
                }
              ]
            }
          }
    </script> </d-front-matter> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> The Whispers of Biology </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item active"> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item dropdown "> <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">submenus </a> <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown"> <a class="dropdown-item " href="/publications/">publications</a> <div class="dropdown-divider"></div> <a class="dropdown-item " href="/projects/">projects</a> <div class="dropdown-divider"></div> <a class="dropdown-item " href="/blog/">blog</a> </div> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link">ctrl k <i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="post distill"> <d-title> <h1>What are GO, MSigDB, KEGG, ORA, and GSEA?</h1> <p>What are those terms in transcriptomics analysis?</p> </d-title> <d-byline></d-byline> <d-article> <d-contents> <nav class="l-text figcaption"> <h3>Contents</h3> <div> <a href="#let-s-start-with-the-full-name-of-each">Let's start with the full name of each</a> </div> <div> <a href="#grouping-according-to-the-usage">Grouping according to the usage</a> </div> <div> <a href="#summary-of-the-connections">Summary of the connections</a> </div> </nav> </d-contents> <h2 id="lets-start-with-the-full-name-of-each">Letâ€™s start with the full name of each</h2> <ol> <li>Gene Ontology (GO)</li> <li>Molecular Signatures Database (MSigDB)</li> <li>Kyoto Encyclopedia of Genes and Genomes (KEGG)</li> <li>Over-Representation Analysis (ORA)</li> <li>Gene Set Enrichment Analysis (GSEA)</li> </ol> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/post2024/GSEAcover-480.webp 480w,/assets/img/post2024/GSEAcover-800.webp 800w,/assets/img/post2024/GSEAcover-1400.webp 1400w," type="image/webp" sizes="95vw"></source> <img src="/assets/img/post2024/GSEAcover.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <div class="caption"> Picture generated by Gemini, Prompt by <a href="https://kuchikinamthip.github.io/" target="_blank">Krittiyabhorn Kongtanawanich.</a> </div> <hr> <h2 id="grouping-according-to-the-usage">Grouping according to the usage</h2> <h3 id="1-source-of-gene-set---go-msigdb-and-kegg">1. Source of gene set -&gt; GO, msigDB, and KEGG</h3> <h4 id="11-gene-ontology-go">1.1 <a href="http://geneontology.org/" rel="external nofollow noopener" target="_blank">Gene Ontology (GO)</a> </h4> <ul> <li>GO provides a structured vocabulary for annotating genes and gene products based on their biological processes, cellular components, and molecular functions. It is widely used to categorize genes into meaningful biological contexts.</li> <li>GO includes BP, CC and MF</li> </ul> <h4 id="12-kyoto-encyclopedia-of-genes-and-genomes-kegg">1.2 <a href="https://www.kegg.jp/kegg/" rel="external nofollow noopener" target="_blank">Kyoto Encyclopedia of Genes and Genomes (KEGG)</a> </h4> <ul> <li>KEGG is a database that provides information on biological pathways, diseases, and drugs. It includes curated pathways that represent molecular interactions and reactions in various biological systems. KEGG pathways are often used in conjunction with GO for pathway analysis.</li> <li>KEGG includes pathway, module, enzyme, network, drug and disease</li> </ul> <h4 id="13-molecular-signatures-database-msigdb">1.3 <a href="https://www.gsea-msigdb.org/gsea/msigdb/" rel="external nofollow noopener" target="_blank">Molecular Signatures Database (MSigDB)</a> </h4> <p>MSigDB is a collection of annotated gene sets that can be used for gene set enrichment analyses. It includes gene sets derived from various sources, including GO terms, KEGG pathways, and other curated datasets. Researchers use MSigDB to access predefined gene sets for enrichment analyses.</p> <h4 id="14-other-sources-of-gene-set">1.4 Other sources of gene set</h4> <p>Other sources of gene set:</p> <ul> <li><a href="https://wikipathways.org/" rel="external nofollow noopener" target="_blank">WikiPathway</a></li> <li><a href="https://maayanlab.cloud/Enrichr/" rel="external nofollow noopener" target="_blank">EnrichrDb</a></li> <li><a href="https://reactome.org/" rel="external nofollow noopener" target="_blank">Reactome</a></li> <li><a href="https://www.ncbi.nlm.nih.gov/mesh/" rel="external nofollow noopener" target="_blank">MeSH</a></li> <li><a href="https://www.disgenet.org/" rel="external nofollow noopener" target="_blank">DisGeNET</a></li> <li><a href="https://disease-ontology.org/" rel="external nofollow noopener" target="_blank">Disease Ontology (DO)</a></li> <li> <a href="http://ncg.kcl.ac.uk/" rel="external nofollow noopener" target="_blank">Network of Cancer Gene (NCG)</a> (version 6 and v7)</li> <li> <a href="https://maayanlab.cloud/covid19/" rel="external nofollow noopener" target="_blank">The COVID-19 Drug and Gene Set Library</a>.</li> </ul> <hr> <h3 id="2-application-in-analyses---ora-and-gsea">2. Application in Analyses -&gt; ORA and GSEA</h3> <h4 id="21-over-representation-analysis-ora">2.1 Over-Representation Analysis (ORA)</h4> <ul> <li>ORA is a statistical method used to determine if a set of genes (e.g., differentially expressed genes) contains a higher proportion of genes associated with specific biological functions or pathways than would be expected by chance. This method often uses databases like GO and KEGG to assess enrichment.</li> <li>In ORA, researchers might input a list of differentially expressed genes to see if they are over-represented in specific GO terms or KEGG pathways.</li> </ul> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/post2024/ORA-480.webp 480w,/assets/img/post2024/ORA-800.webp 800w,/assets/img/post2024/ORA-1400.webp 1400w," type="image/webp" sizes="95vw"></source> <img src="/assets/img/post2024/ORA.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <div class="caption"> How the genes in ORA were defined. <a href="https://ntuhmc.ntuh.gov.tw/epaper-54th.htm" target="_blank" rel="external nofollow noopener">Credit pic</a> </div> <h4 id="22-gene-set-enrichment-analysis-gsea">2.2 Gene Set Enrichment Analysis (GSEA)</h4> <ul> <li>GSEA evaluates whether predefined sets of genes show statistically significant differences between two biological states (e.g., treated vs. untreated). Unlike ORA, GSEA considers all genes in the dataset and ranks them based on expression levels, allowing it to detect subtle changes across entire gene sets.</li> <li>In GSEA, the entire list of ranked genes can be assessed against the same or similar sets from msigDB, allowing for a broader understanding of the biological implications of expression changes.</li> <li>For more information regarding GSEA issue, please read <a href="https://github.com/crazyhottommy/RNA-seq-analysis/blob/master/GSEA_explained.md" rel="external nofollow noopener" target="_blank">a post from Ming (Tommy) Tang</a>.</li> </ul> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/post2024/ORAvsGSEA-480.webp 480w,/assets/img/post2024/ORAvsGSEA-800.webp 800w,/assets/img/post2024/ORAvsGSEA-1400.webp 1400w," type="image/webp" sizes="95vw"></source> <img src="/assets/img/post2024/ORAvsGSEA.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <div class="caption"> Overview of GSEA. <a href="https://github.com/crazyhottommy/RNA-seq-analysis/blob/master/GSEA_explained.md" target="_blank" rel="external nofollow noopener">Credit pic</a> </div> <hr> <h2 id="summary-of-the-connections">Summary of the connections</h2> <blockquote> <p>In summary, GO provides functional annotations, KEGG offers pathway information, msigDB aggregates these resources into usable gene sets, while ORA and GSEA are methodologies that leverage these databases to interpret gene expression data in a biological context.</p> </blockquote> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/post2024/ORAvsGSEA-480.webp 480w,/assets/img/post2024/ORAvsGSEA-800.webp 800w,/assets/img/post2024/ORAvsGSEA-1400.webp 1400w," type="image/webp" sizes="95vw"></source> <img src="/assets/img/post2024/ORAvsGSEA.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <div class="caption"> The different between ORA and GSEA. <a href="https://ntuhmc.ntuh.gov.tw/epaper-54th.htm" target="_blank" rel="external nofollow noopener">Credit pic</a> </div> </d-article> <d-appendix> <d-footnote-list></d-footnote-list> <d-citation-list></d-citation-list> </d-appendix> <d-bibliography src="/assets/bibliography/2018-12-22-distill.bib"></d-bibliography> <div id="giscus_thread" style="max-width: 930px; margin: 0 auto;"> <script>
      let giscusTheme = determineComputedTheme();
      let giscusAttributes = {
        src: 'https://giscus.app/client.js',
        'data-repo': 'KuchikiNamthip/KuchikiNamthip.github.io',
        'data-repo-id': 'R_kgDOMFy1BA',
        'data-category': 'Comments',
        'data-category-id': '',
        'data-mapping': 'title',
        'data-strict': '1',
        'data-reactions-enabled': '1',
        'data-emit-metadata': '0',
        'data-input-position': 'bottom',
        'data-theme': giscusTheme,
        'data-lang': 'en',
        crossorigin: 'anonymous',
        async: '',
      };

      let giscusScript = document.createElement('script');
      Object.entries(giscusAttributes).forEach(([key, value]) => giscusScript.setAttribute(key, value));
      document.getElementById('giscus_thread').appendChild(giscusScript);
    </script> <noscript>Please enable JavaScript to view the <a href="http://giscus.app/?ref_noscript" rel="external nofollow noopener" target="_blank">comments powered by giscus.</a> </noscript> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> Â© Copyright 2025 Krittiyabhorn (Namthip) Kongtanawanich. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script src="/assets/js/mathjax-setup.js?70d799092f862ad98c7876aa47712e20"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script defer src="/assets/js/progress-bar.js?2f30e0e6801ea8f5036fa66e1ab0a71a" type="text/javascript"></script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>
    addBackToTop();
  </script> <script type="module" src="/assets/js/search/ninja-keys.min.js?a3446f084dcaecc5f75aa1757d087dcf"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script src="/assets/js/search-setup.js?6c304f7b1992d4b60f7a07956e52f04a"></script> <script src="/assets/js/search-data.js"></script> <script src="/assets/js/shortcut-key.js?6f508d74becd347268a7f822bca7309d"></script> </body> </html>