---
layout: distill
title: What are GO, MSigDB, KEGG, ORA, and GSEA?
description: What are those terms in transcriptomics analysis?
tags: R GSEA ORA RNA-seq
giscus_comments: true
date: 2024-09-18
related_posts: false
categories: Transcriptomics
featured: false
thumbnail: assets/img/post2024/GSEAcover.png

authors:
  - name: Krittiyabhorn (Namthip) Kongtanawanich
    url: "https://kuchikinamthip.github.io/"
    affiliations:
      name: Faculty of Medicine Siriraj Hospital, Mahidol University, Thailand

bibliography: 2018-12-22-distill.bib

toc:
  - name: Let's start with the full name of each
    # if a section has subsections, you can add them as follows:
    # subsections:
    #   - name: Example Child Subsection 1
    #   - name: Example Child Subsection 2
  - name: Grouping according to the usage
  - name: Summary of the connections
---

## Let's start with the full name of each

1.  Gene Ontology (GO)
2.  Molecular Signatures Database (MSigDB)
3.  Kyoto Encyclopedia of Genes and Genomes (KEGG)
4.  Over-Representation Analysis (ORA)
5.  Gene Set Enrichment Analysis (GSEA)
<div class="row">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.liquid loading="eager" path="assets/img/post2024/GSEAcover.png" title="example image" class="img-fluid rounded z-depth-1" %}
    </div>
</div>
<div class="caption">
    Picture generated by Gemini, Prompt by <a href="https://kuchikinamthip.github.io/" target="_blank">Krittiyabhorn Kongtanawanich.</a>
</div>

---

## Grouping according to the usage

### 1. Source of gene set -> GO, msigDB, and KEGG

#### 1.1 [Gene Ontology (GO)](http://geneontology.org/)

- GO provides a structured vocabulary for annotating genes and gene products based on their biological processes, cellular components, and molecular functions. It is widely used to categorize genes into meaningful biological contexts.
- GO includes BP, CC and MF

#### 1.2 [Kyoto Encyclopedia of Genes and Genomes (KEGG)](https://www.kegg.jp/kegg/)

- KEGG is a database that provides information on biological pathways, diseases, and drugs. It includes curated pathways that represent molecular interactions and reactions in various biological systems. KEGG pathways are often used in conjunction with GO for pathway analysis.
- KEGG includes pathway, module, enzyme, network, drug and disease

#### 1.3 [Molecular Signatures Database (MSigDB)](https://www.gsea-msigdb.org/gsea/msigdb/)

MSigDB is a collection of annotated gene sets that can be used for gene set enrichment analyses. It includes gene sets derived from various sources, including GO terms, KEGG pathways, and other curated datasets. Researchers use MSigDB to access predefined gene sets for enrichment analyses.

#### 1.4 Other sources of gene set

Other sources of gene set:

- [WikiPathway](https://wikipathways.org/)
- [EnrichrDb](https://maayanlab.cloud/Enrichr/)
- [Reactome](https://reactome.org/)
- [MeSH](https://www.ncbi.nlm.nih.gov/mesh/)
- [DisGeNET](https://www.disgenet.org/)
- [Disease Ontology (DO)](https://disease-ontology.org/)
- [Network of Cancer Gene (NCG)](http://ncg.kcl.ac.uk/) (version 6 and v7)
- [The COVID-19 Drug and Gene Set Library](https://maayanlab.cloud/covid19/).

---

### 2. Application in Analyses -> ORA and GSEA

#### 2.1 Over-Representation Analysis (ORA)

- ORA is a statistical method used to determine if a set of genes (e.g., differentially expressed genes) contains a higher proportion of genes associated with specific biological functions or pathways than would be expected by chance. This method often uses databases like GO and KEGG to assess enrichment.
- In ORA, researchers might input a list of differentially expressed genes to see if they are over-represented in specific GO terms or KEGG pathways.
<div class="row">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.liquid loading="eager" path="assets/img/post2024/ORA.png" title="example image" class="img-fluid rounded z-depth-1" %}
    </div>
</div>
<div class="caption">
    How the genes in ORA were defined. <a href="https://ntuhmc.ntuh.gov.tw/epaper-54th.htm" target="_blank">Credit pic</a>
</div>

#### 2.2 Gene Set Enrichment Analysis (GSEA)

- GSEA evaluates whether predefined sets of genes show statistically significant differences between two biological states (e.g., treated vs. untreated). Unlike ORA, GSEA considers all genes in the dataset and ranks them based on expression levels, allowing it to detect subtle changes across entire gene sets.
- In GSEA, the entire list of ranked genes can be assessed against the same or similar sets from msigDB, allowing for a broader understanding of the biological implications of expression changes.
- For more information regarding GSEA issue, please read [a post from Ming (Tommy) Tang](https://github.com/crazyhottommy/RNA-seq-analysis/blob/master/GSEA_explained.md).
<div class="row">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.liquid loading="eager" path="assets/img/post2024/ORAvsGSEA.png" title="example image" class="img-fluid rounded z-depth-1" %}
    </div>
</div>
<div class="caption">
    Overview of GSEA. <a href="https://github.com/crazyhottommy/RNA-seq-analysis/blob/master/GSEA_explained.md" target="_blank">Credit pic</a>
</div>

---

## Summary of the connections

> In summary, GO provides functional annotations, KEGG offers pathway information, msigDB aggregates these resources into usable gene sets, while ORA and GSEA are methodologies that leverage these databases to interpret gene expression data in a biological context.

<div class="row">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.liquid loading="eager" path="assets/img/post2024/ORAvsGSEA.png" title="example image" class="img-fluid rounded z-depth-1" %}
    </div>
</div>
<div class="caption">
    The different between ORA and GSEA. <a href="https://ntuhmc.ntuh.gov.tw/epaper-54th.htm" target="_blank">Credit pic</a>
</div>
